http://git.savannah.gnu.org/gitweb/?p=patch.git;a=commitdiff;h=f376c5db4a4b169176996c67c8c5ac53c3b18a44

From f376c5db4a4b169176996c67c8c5ac53c3b18a44 Mon Sep 17 00:00:00 2001
From: Andreas Gruenbacher <agruen@suse.de>
Date: Fri, 15 Jan 2010 19:24:59 +0100
Subject: [PATCH] Fix HAVE_DECL_STRNLEN and HAVE_DECL_STRNDUP checks

* gl/lib/strndup.c: HAVE_DECL_STRNLEN is either 0 or 1 so we need
to test for it with #if, not #ifdef.
* gl/lib/xstrndup.c: Likewise for HAVE_DECL_STRNDUP.
---
 ChangeLog         |    4 ++++
 gl/lib/strndup.c  |    2 +-
 gl/lib/xstrndup.c |    2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/gl/lib/strndup.c b/gl/lib/strndup.c
index 6548196..6fdc442 100644
--- a/gl/lib/strndup.c
+++ b/gl/lib/strndup.c
@@ -23,7 +23,7 @@
 
 #include <stdlib.h>
 
-#ifndef HAVE_DECL_STRNLEN
+#if !HAVE_DECL_STRNLEN
 extern size_t strnlen (const char *, size_t);
 #endif
 
diff --git a/gl/lib/xstrndup.c b/gl/lib/xstrndup.c
index 6f16445..b0e5495 100644
--- a/gl/lib/xstrndup.c
+++ b/gl/lib/xstrndup.c
@@ -23,7 +23,7 @@
 #include <string.h>
 #include "xalloc.h"
 
-#ifndef HAVE_DECL_STRNDUP
+#if !HAVE_DECL_STRNDUP
 extern char *strndup (const char *, size_t);
 #endif
 
-- 
1.6.5

